### PR TITLE
Logging types and subtypes

### PR DESCRIPTION
**Pull request by renefritze**
This implements the proposal in https://github.com/readthedocs/sphinx-autoapi/issues/275

I'm not very sure about the subtypes. 

> "Unknown type" should possibly be an error so leave that one out for now. "Invalid all entry" could be invalid_sources.

Didn't come across either of those.

### CODE REVIEW

File: autoapi/mappers/dotnet.py [hunk 1]
```diff
@@ -104,9 +104,16 @@ def load(self, patterns, dirs, ignore=None, **kwargs):
                 )
                 _, error_output = proc.communicate()
                 if error_output:
-                    LOGGER.warning(error_output)
+                    LOGGER.warning(
+                        error_output, type="autoapi", subtype="metadata_generation"
```
Comment (AWhetter) [inline autoapi/mappers/dotnet.py:108]: Let's treat this as a `not_readable`. Similarly for the warning below as well.

**Review by AWhetter — CHANGES_REQUESTED** [review]
Thanks for making this change. I've made a few suggestions. Let me know what you think.

File: autoapi/mappers/python/mapper.py [hunk 2]
```diff
@@ -359,15 +363,19 @@ def create_class(self, data, options=None, **kwargs):
...
+            )
         else:
             obj = cls(
                 data,
                 class_content=self.app.config.autoapi_python_class_content,
                 options=self.app.config.autoapi_options,
                 jinja_env=self.jinja_env,
                 app=self.app,
-                **kwargs
+                **kwargs,
```
[the remark this answers is not in the document]
Comment (renefritze) [reply autoapi/mappers/python/mapper.py:378]: Yeah. I didn't touch that myself.

File: docs/reference/config.rst [hunk 3]
```diff
@@ -276,3 +276,35 @@ Advanced Options
...
+   ``".subtype"`` can be left out to cover all subtypes. To suppress all AutoAPI
+   warnings add the type ``"autoapi"`` to the list:
+
+   .. code-block:: python
+
+      suppress_warnings = ["autoapi"]
+
+   If narrower suppression is wanted, the available subtypes for AutoAPI are:
+
+     * python_import_resolution
```
[the remark this answers is not in the document]
Comment (renefritze) [reply docs/reference/config.rst:299]: That would be good, I agree. I've tried to add some now, but they feel somewhat clunky to me.

**Review by AWhetter — APPROVED** [review]
Looks good to me. Thanks for the changes!

**Comment by renefritze** [comment]
Thank you for merging this. Would be great to get a new release soonish so I can start to use this myself.

**Comment by AWhetter** [comment]
v1.8.0 was released with this in it.

**Comment by renefritze** [comment]
Awesome. Thank you so much!

**Pull request merged**